queue_job/models/queue_job.py

```diff
@@ -308,13 +308,16 @@ def _change_job_state(self, state, result=None):
             if state == DONE:
                 job_.set_done(result=result)
                 job_.store()
+                self.env.cr.commit()
```

Comment:
why commit? `record.env["queue.job"].flush()` should be enough then

Reply:
commit should be a "[FIX]"

Reply:
Thank nils, then PR is updated

Reply:
Added tests also